packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/internal/UserImpl.kt

```diff
@@ -17,9 +17,68 @@
 package io.realm.mongodb.internal
 
 import io.realm.internal.interop.NativePointer
+import io.realm.internal.interop.RealmInterop
+import io.realm.internal.interop.channelResultCallback
+import io.realm.internal.interop.sync.CoreUserState
+import io.realm.internal.platform.freeze
+import io.realm.internal.util.use
 import io.realm.mongodb.User
+import kotlinx.coroutines.channels.Channel
 
 internal class UserImpl(
     val nativePointer: NativePointer,
-    val app: AppImpl
-) : User
+    override val app: AppImpl
+) : User {
+
+    override val state: User.State
+        get() = UserImpl.fromCoreState(RealmInterop.realm_user_get_state(nativePointer))
+
+    // TODO Can maybe fail, but we could also cache the return value?
+    override val identity: String
+        get() = RealmInterop.realm_user_get_identity(nativePointer)
+
+    override val loggedIn: Boolean
+        get() = RealmInterop.realm_user_is_logged_in(nativePointer)
+
+    override suspend fun logOut() {
+        Channel<Result<Unit>>(1).use { channel ->
+            RealmInterop.realm_app_log_out(
+                app.nativePointer,
+                nativePointer,
+                channelResultCallback<Unit, Unit>(channel) {
+                    // No-op
+                }.freeze()
+            )
+        }
```

Comment:
I guess we still need some suspending call until the underlying operation is done, so that we don't return before the operation has actually been performed.